packages/flutter/lib/src/material/typography.dart

```diff
@@ -6,623 +6,449 @@ import 'package:flutter/foundation.dart';
-///  * [Theme], for other aspects of a material design application that can be
-///    globally adjusted, such as the color scheme.
-///  * <http://material.google.com/style/typography.html>
+///  * [ThemeData.typography], which can be used to configure the
+///    text themes used to create [ThemeData.textTheme],
+///    [ThemeData.primaryTxtTheme], [ThemeData.accentTextTheme].
+///  * <https://material.io/design/typography/>
 @immutable
-class TextTheme extends Diagnosticable {
-  /// Creates a text theme that uses the given values.
-  ///
-  /// Rather than creating a new text theme, consider using [Typography.black]
-  /// or [Typography.white], which implement the typography styles in the
-  /// material design specification:
+class Typography extends Diagnosticable {
+  /// Creates a typography instance.
   ///
-  /// <https://material.google.com/style/typography.html#typography-styles>
+  /// If [platform] is specified, the default values for [black] and [white]
+  /// are [blackCupertino] and [whiteCupertino] respectively. Otherwise
+  /// they are [blackMountainView] and [whiteMoutainView].
   ///
-  /// If you do decide to create your own text theme, consider using one of
-  /// those predefined themes as a starting point for [copyWith] or [apply].
-  const TextTheme({
-    this.display4,
-    this.display3,
-    this.display2,
-    this.display1,
-    this.headline,
-    this.title,
-    this.subhead,
-    this.body2,
-    this.body1,
-    this.caption,
-    this.button,
-  });
-
-  /// Extremely large text.
-  ///
-  /// The font size is 112 pixels.
-  final TextStyle display4;
-
-  /// Very, very large text.
-  ///
-  /// Used for the date in the dialog shown by [showDatePicker].
-  final TextStyle display3;
-
-  /// Very large text.
-  final TextStyle display2;
-
-  /// Large text.
-  final TextStyle display1;
-
-  /// Used for large text in dialogs (e.g., the month and year in the dialog
-  /// shown by [showDatePicker]).
-  final TextStyle headline;
-
-  /// Used for the primary text in app bars and dialogs (e.g., [AppBar.title]
-  /// and [AlertDialog.title]).
-  final TextStyle title;
-
-  /// Used for the primary text in lists (e.g., [ListTile.title]).
-  final TextStyle subhead;
-
-  /// Used for emphasizing text that would otherwise be [body1].
-  final TextStyle body2;
-
-  /// Used for the default text style for [Material].
-  final TextStyle body1;
-
-  /// Used for auxiliary text associated with images.
-  final TextStyle caption;
+  /// The default values for [englishLike], [dense], and [tall] are
+  /// [englishLike2014], [dense2014], and [tall2014].
+  factory Typography({
+    TargetPlatform platform,
+    TextTheme black,
+    TextTheme white,
+    TextTheme englishLike,
+    TextTheme dense,
+    TextTheme tall,
+  }) {
+    switch (platform) {
+      case TargetPlatform.iOS:
+        black ??= blackCupertino;
+        white ??= whiteCupertino;
+        break;
+      default:
```

Comment:
please be explicit about the platforms rather than using default

Reply:
Done